docs/02_database/05_database-interface/01_entitydb.mdx

```diff
@@ -662,62 +747,417 @@ final var subscription = db.subscribe(Trade.class)
+
+So what does this mean in practice?
+The batching period is always set globally, i.e. for an update queue subscription or on a data server query.
+Never on the individual row level.
+
+The goal of this batching is to minimise work, and to publish updates that are accurate, rather than to publish every change.
+
+In this regard it is better to think in terms of a batching interval, rather than a batching period, which looks something like this:
+
+```
+...
+*
+|---- gathering phase ----| (batching period)
+*
+*
+*
+*
+*
+|---- flushing phase -----| (tick happens here)
+* aggregate updates
+* publish updates
+|---- gathering phase ----| (next batching period)
+*
+*
+...
+```
+
+### Batching is non-deterministic
+
+During our gathering phase, we listen for updates; and during the flushing phase, we aggregate and publish.
+
+Let's look at an example:
+
+Imagine we insert a trade with quantity 100. Within milliseconds, we update this trade and set the quantity to 200.
+
+Depending on when the tick happens, you could either get two updates:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+*
+* insert (qty 100)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 100)
+|---- gathering phase ----|
+* modify (qty 200)
+*
+*
+*
+*
+|---- flushing phase -----|
+* aggregate updates
+* publish updates modify (qty 200)
+|---- gathering phase ----| (batching period)
+*
+*
+```
+
+1. Insert trade, qty = 100
+2. Modify trade, qty = 200
+
+Or you could receive a single update:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+* insert (qty 100)
+* modify (qty 200)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 200)
+```
+
+1. Insert trade, qty = 200
+
+Depending on timing, either case is possible, and the smaller the gap between the insert and modify, the more likely the second scenario becomes.
+However, even if these operations happen immediately after each other, we could still receive two updates.
+
+When we get two updates of the same record within the batching interval we will always aggregate the updates.
+This could even mean that updates will be disregarded all together, for example if we receive an insert and a delete within the batching interval.
+
+This will always be non-deterministic, as we don't provide any guarantees.
+
+#### Batching at different levels
+
+The other thing to note is that batching happens on both the data server and the update queue level.
+We batch on the queue level to minimise the amount of work the data server has to do.
+Because we pre-allocate LMDB when the data sever starts, having too many needless updates would eat into that allocation.
+We batch on the data server to minimise the load on the client and the connection.
+
+### Out-of-sequence updates on the queue
+
+Updates are published on the update queue once a record has been written to the database, or when a transaction completes you are using a transaction database.
```

Comment:
Should this be `if you are using a transactional database` rather than `you are using a transaction database`?

Reply:
Also, I am thinking that as we don't support Aerospike anymore, maybe we don't need to worry too much about what happens in non-transactional databases?